### PR TITLE
Bucket sort

### PR DESCRIPTION
**Pull request by olexandr-maltsev-work**

### CODE REVIEW

**Review by mpaselskyi — APPROVED** [review]

File: src/main/java/com/softserve/algorithms/sort/BucketSort.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                        bucket.set(j - 1, temp);
+                    }
+                }
+            }
+        }
+        return bucket;
+    }
+
+
+    class Bucket {
```
Comment (tania-klueva) [inline src/main/java/com/softserve/algorithms/sort/BucketSort.java]: One class one file

File: src/main/java/com/softserve/algorithms/sort/BucketSort.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    /**
+     * Find bucket index in the input array
+     *
+     * @param input Array
+     * @param number from this array
+     * @return bucket index
+     */
+    int returnBucketIndex(int[] input, int number) {
```
Comment (tania-klueva) [inline src/main/java/com/softserve/algorithms/sort/BucketSort.java]: access can be private

File: src/main/java/com/softserve/algorithms/sort/BucketSort.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        return (int) sortingFunctionResult;
+    }
+
+    /**
+     * If bucket has more than one element, this bucket must be sorted
+     * I choose for bucket sorting Insertion algorithm
+     * @param bucket with integers
+     * @return sorted bucket
+     */
+    ArrayList<Integer> sortBucket(ArrayList<Integer> bucket) {
```
Comment (tania-klueva) [inline src/main/java/com/softserve/algorithms/sort/BucketSort.java]: access can be private

File: src/main/java/com/softserve/algorithms/sort/BucketSort.java (hunk 1 above)
Comment (tania-klueva) [inline src/main/java/com/softserve/algorithms/sort/BucketSort.java]: can be private

**Review by tania-klueva — CHANGES_REQUESTED** [review]
access modifiers

**Pull request closed without merging**